x/feedist/msg.go

```diff
@@ -0,0 +1,62 @@
+package feedist
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave/errors"
+)
+
+const (
+	pathNewRevenueMsg    = "feedist/newrevenue"
+	pathDistributeMsg    = "feedist/distribute"
+	pathUpdateRevenueMsg = "feedist/updateRevenue"
+)
+
+func (msg *NewRevenueMsg) Validate() error {
+	if err := msg.Admin.Validate(); err != nil {
+		return errors.Wrap(err, "invalid admin address")
+	}
+	if len(msg.Recipients) == 0 {
+		return errors.InvalidMsgErr.New("at least one recipient must be given")
+	}
+	for i, r := range msg.Recipients {
+		if err := r.Address.Validate(); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("recipient %d address", i))
+		}
+		if r.Weight <= 0 {
```

Comment:
req: sum of weights must not be > MaxInt64

Reply:
How can this be checked if I am using int64 for storing the sum?

Reply:
I would rather prefer a fix limit for both in the code. That's easier to communicate. Or we must deal with bigInts in the distribution code.

Reply:
I think you could use validation logic on reset revenue or init to validate total weight. Then we are secure in distribution

Reply:
Implemented in https://github.com/iov-one/weave/pull/316/commits/7f699ed7f68580c219ff635f4cd6dc15f5c17b86